### DISCUSSION TITLE
Can I use Zed's auto-formatting programmatically?

### DISCUSSION THREAD

**Discussion by bobbykjack** [zed-industries/zed#17634] [Feature Requests]
I like zed's auto-formatting, and I want all files in a project to use it. However, some files are generated programatically, so unless I manually open each file in zed, make a 'null change', and save it, the formatting gets out of sync.

Is there a way I can do something like:

`$ zed --auto-format index.html`

?

**Comment by SomeoneToIgnore** [comment]
There's no such functionality yet.
Also, the formatting is not really done by Zed, but rather prettier or a corresponding language server, so it might be more straightforward to call them instead.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
